includes/transformer/class-transformers-manager.php

```diff
@@ -0,0 +1,292 @@
+		 *
+		 * @since version_number_transformer_management_placeholder
+		 *
+		 * @param bool $should_register Should the ActivityPub transformer be registered. Default is `true`.
+		 * @param \ActivityPub\Transformer\Base $transformer_instance Widget instance.
+		 */
+		// TODO: does not implementing this slow down the website? -> compare with gutenberg block registration.
+		// $should_register = apply_filters( 'activitypub/transformers/is_transformer_enabled', true, $transformer_instance );
+
+		// if ( ! $should_register ) {
+		//  return false;
+		// }
+
+		$this->transformers[ $transformer_name ] = $transformer_instance;
+
+		return true;
+	}
+
+	/**
+	 * Init transformers.
+	 *
+	 * Initialize ActivityPub transformer manager.
+	 * Include the builtin transformers by default and add third party ones.
+	 *
+	 * @since version_number_transformer_management_placeholder
+	 * @access private
+	 */
+	private function init_transformers() {
+		$builtin_transformers = [
+			'post',
+		];
+
+		$this->transformers = [];
+
+		foreach ( $builtin_transformers as $transformer_name ) {
+			include ACTIVITYPUB_PLUGIN_DIR . 'includes/transformer/class-' . $transformer_name . '.php';
+
+			$class_name = ucfirst( $transformer_name );
+
+			$class_name = '\Activitypub\Transformer\\' . $class_name;
+
+			$this->register( new $class_name() );
+		}
+
+		/**
+		 * Let other transformers register.
+		 *
+		 * Fires after the built-in Activitypub transformers are registered.
+		 *
+		 * @since version_number_transformer_management_placeholder
+		 *
+		 * @param Transformers_Manager $this The widgets manager.
+		 */
+		do_action( 'activitypub_transformers_register', $this );
+	}
+
+	/**
+	 * Get available ActivityPub transformers.
+	 *
+	 * Retrieve the registered transformers list. If given a transformer name
+	 * it returns the given transformer if it is registered.
+	 *
+	 * @since version_number_transformer_management_placeholder
+	 * @access public
+	 *
+	 * @param string $transformer_name Optional. Transformer name. Default is null.
+	 *
+	 * @return Base|Base[]|null Registered transformers.
+	 */
+	public function get_transformers( $transformer_name = null ) {
+		if ( is_null( $this->transformers ) ) {
+			$this->init_transformers();
+		}
+
+		if ( null !== $transformer_name ) {
+			return isset( $this->transformers[ $transformer_name ] ) ? $this->transformers[ $transformer_name ] : null;
+		}
+
+		return $this->transformers;
+	}
+
+	/**
+	 * Get the mapped ActivityPub transformer.
+	 *
+	 * Returns a new instance of the needed WordPress to ActivityPub transformer.
+	 *
+	 * @since version_number_transformer_management_placeholder
+	 * @access public
+	 *
+	 * @param WP_Post|WP_Comment $object The WordPress Post/Comment.
+	 *
+	 * @return \ActivityPub\Transformer\Base|null Registered transformers.
+	 */
+	public function get_transformer( $object ) {
+		switch ( get_class( $object ) ) {
+			case 'WP_Post':
+				$post_type = get_post_type( $object );
```

Comment:
this looks a bit complicated and redundant. the way you implemented it, does not require a switch/case, I think.